### ISSUE TITLE
raft: `fsm_output::empty()` definition incomplete?

### ISSUE THREAD

**Issue by kbr-** [scylladb/scylladb#10912]
```cpp
struct fsm_output {
    struct applied_snapshot {
        snapshot_descriptor snp;
        bool is_local;
        snapshot_id prev_snp_id;
    };
    std::optional<std::pair<term_t, server_id>> term_and_vote;
    std::vector<log_entry_ptr> log_entries;
    std::vector<std::pair<server_id, rpc_message>> messages;
    // Entries to apply.
    std::vector<log_entry_ptr> committed;
    std::optional<applied_snapshot> snp;
    // Latest configuration obtained from the log in case it has changed
    // since last fsm output poll.
    std::optional<server_address_set> configuration;
    std::optional<read_id> max_read_id_with_quorum;
    // Set to true if a leadership transfer was aborted since the last output
    bool abort_leadership_transfer;

    // True if there is no new output
    bool empty() const {
        return !term_and_vote &&
            log_entries.size() == 0 && messages.size() == 0 &&
            committed.size() == 0 && !snp &&
            !configuration;
    }
};
```

It looks like when `std::optional<read_id> max_read_id_with_quorum` field was added, `empty()` should have been extended as well but wasn't.

**Comment by gleb-cloudius** [comment]
May be. `empty()` is used by tests only and for not test we have entirely different condition to detect the non empty output: https://github.com/scylladb/scylla/blob/f5ff687b64583175fc4e6ce65a4cb4dfb77eaf64/raft/fsm.cc#L288. @kostja as the one who added `empty()` why are they so different?

**Comment by kostja** [comment]
Because no one bothered to update it and since it's only used in tests and no more tests for it were added as the definition bloat happened it didn't fire.

**Comment by harsh020** [comment]
Hi @kostja, I was hoping to start contributing to scylladb. This seem like a very simple issue to get started. Can I send out a PR to fix the same?

**Comment by kostja** [comment]
no, nobody is working on it.

**Comment by harsh020** [comment]
@Azanul afaik, the PR fixes the issue and was queued to merge.

**Comment by kbr-scylla** [comment]
Ah yes, we forgot about the `Fixes: #10912`.
